src/umockcallrecorder.c

```diff
@@ -269,82 +377,89 @@ const char* umockcallrecorder_get_expected_calls(UMOCKCALLRECORDER_HANDLE umock_
-                    char* stringified_call = umockcall_stringify(umock_call_recorder->expected_calls[i].umockcall);
-                    if (stringified_call == NULL)
-                    {
-                        /* Codes_SRS_UMOCKCALLRECORDER_01_030: [ If umockcall_stringify fails, umockcallrecorder_get_expected_calls shall fail and return NULL. ]*/
-                        break;
-                    }
-                    else
+                    /* Codes_SRS_UMOCKCALLRECORDER_01_056: [ If umockcall_get_ignore_all_calls returns a negative value then umockcallrecorder_get_expected_calls shall fail and return NULL. ]*/
+                    UMOCK_LOG("umockcallrecorder: Cannot get the ignore_all_calls flag.");
+                    break;
+                }
+                else
+                {
+                    if ((umock_call_recorder->expected_calls[i].is_matched == 0) &&
+                        /* Codes_SRS_UMOCKCALLRECORDER_01_054: [ Calls that have the ignore_all_calls property set shall not be reported in the expected call list. ]*/
+                        (ignore_all_calls == 0))
                     {
-                        size_t stringified_call_length = strlen(stringified_call);
-                        new_expected_calls_string = (char*)umockalloc_realloc(umock_call_recorder->expected_calls_string, current_length + stringified_call_length + 1);
-                        if (new_expected_calls_string == NULL)
+                        /* Codes_SRS_UMOCKCALLRECORDER_01_028: [ The string for each call shall be obtained by calling umockcall_stringify. ]*/
+                        char* stringified_call = umockcall_stringify(umock_call_recorder->expected_calls[i].umockcall);
+                        if (stringified_call == NULL)
                         {
-                            umockalloc_free(stringified_call);
-
-                            /* Codes_SRS_UMOCKCALLRECORDER_01_031: [ If allocating memory for the resulting string fails, umockcallrecorder_get_expected_calls shall fail and return NULL. ]*/
+                            /* Codes_SRS_UMOCKCALLRECORDER_01_030: [ If umockcall_stringify fails, umockcallrecorder_get_expected_calls shall fail and return NULL. ]*/
                             break;
                         }
                         else
                         {
-                            umock_call_recorder->expected_calls_string = new_expected_calls_string;
-                            (void)memcpy(umock_call_recorder->expected_calls_string + current_length, stringified_call, stringified_call_length + 1);
-                            current_length += stringified_call_length;
-                        }
+                            size_t stringified_call_length = strlen(stringified_call);
+                            new_expected_calls_string = umockalloc_realloc(umock_call_recorder->expected_calls_string, current_length + stringified_call_length + 1);
```

Comment:
>new_expected_calls_string = umockalloc_realloc [](start = 28, length = 46)

smells like exclusive locking should be used...

Reply:
You are right

---
In reply to: [530653997](https://github.com/Azure/umock-c/pull/177#discussion_r530653997) [](ancestors = 530653997)